File: dataset/human/python/file_1308.py
from collections import Counter,defaultdict,deque
from heapq import heappop,heappush,heapify
from bisect import bisect_left,bisect_right 
import sys,math,itertools,fractions,pprint
sys.setrecursionlimit(10**8)
mod = 10**9+7
INF = float('inf')
def inp(): return int(sys.stdin.readline())
def inpl(): return list(map(int, sys.stdin.readline().split()))

n = inp()
a = inpl()
su = sum(a)
acc = list(itertools.accumulate(a))
b = [abs((su-acc[i]) - acc[i]) for i in range(n-1)]
# print(b)
print(min(b))

File: dataset/human/java/file_1308.java
import java.util.*;

class Main{
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] a = new int[n];
        for(int i = 0; i < n; i++){
            a[i] = sc.nextInt();
        }

        long ans = Long.MAX_VALUE;
        long x = a[0];
        long y = 0;
        for(int i = 1; i < n; i++){
            y += a[i];
        }
        ans = Math.min(ans, Math.abs(x-y));
        if(n > 2) {
            for (int i = 1; i < n-1; i++) {
                x += a[i];
                y -= a[i];
                ans = Math.min(ans, Math.abs(x - y));
            }
        }
        System.out.println(ans);
    }
}
